python/ray/serve/tests/test_deployment_scheduler.py

```diff
@@ -0,0 +1,263 @@
+import sys
+
+import pytest
+
+import ray
+from ray.tests.conftest import *  # noqa
+from ray.serve._private.deployment_scheduler import (
+    DeploymentScheduler,
+    SpreadDeploymentSchedulingPolicy,
+    DriverDeploymentSchedulingPolicy,
+    ReplicaSchedulingRequest,
+    DeploymentDownscaleRequest,
+)
+
+
+@ray.remote(num_cpus=1)
+class Replica:
+    def get_node_id(self):
+        return ray.get_runtime_context().get_node_id()
+
+
+def test_spread_deployment_scheduling_policy_upscale(ray_start_cluster):
```

Comment:
Could you add a test where you create `Replicas` that spawn other tasks and actors? We should check that the `DeploymentScheduler` schedules around other them as expected.

Reply:
What do you want to test? Deployment scheduler just uses "SPREAD" strategy from ray core so it doesn't have any special handling of other tasks or actors.

Reply:
The test is moreso to prevent regressions. I want to make the scheduler is aware of other tasks and actors. I was thinking you could place a task that consumes 1 CPU on a node with only 1 CPU and make sure the scheduler doesn't try to schedule any replicas on it.

Reply:
Gotcha, that sounds good.